Documentation/authentication.md

```diff
@@ -0,0 +1,164 @@
+# Authentication Guide of etcd v3
+
+## Overview
+
+Authentication was added in etcd 2.1. In the v3 API, the authentication feature, its API and user interface, was modified slightly. This guide will help users set up basic authentication in etcd v3.
+
+## Special Users and Roles
+
+There is one special user, `root`, and there are two special roles, `root` and `guest`.
+
+### User `root`
+
+User `root` must be created before auth can be activated. It must have the `root` role and is allowed to change anything inside etcd. The idea behind the `root` user is for administrative purposes: managing other ordinal users and roles.  In addition, the root role is required for other management functionalities including adding/removing/updating a node in a cluster, defragmentation, taking a snapshot, and checking endpoint health.
+
+### Role `root`
+
+Role `root` can be granted to any user other than root. Having access via the root role not only allows global read-write access (as was the case before 2.1) but allows modification of the authentication policy and all administrative things, like modifying the cluster membership.
```

Comment:
The role `root` may be granted to any user, in addition to the root user. A user with the root role has both global read-write access and permission to update the cluster's authentication configuration. Furthermore, the root role grants privileges for general cluster maintenance, including modifying cluster membership, defragmenting the store, and taking snapshots.
?